examples/platforms/efr32/radio.c

```diff
@@ -0,0 +1,864 @@
+    (void)aInstance;
+
+    CORE_DECLARE_IRQ_STATE;
+    CORE_ENTER_CRITICAL();
+
+    if (sState == kStateSleep || sState == kStateReceive)
+    {
+        otLogInfoPlat(sInstance, "State=kStateSleep", NULL);
+        error = kThreadError_None;
+        sState = kStateSleep;
+
+        if (sIsReceiverEnabled)
+        {
+            RAIL_RfIdleExt(RAIL_IDLE, true);
+            sIsReceiverEnabled = false;
+        }
+    }
+
+    CORE_EXIT_CRITICAL();
+    return error;
+}
+
+ThreadError otPlatRadioReceive(otInstance *aInstance, uint8_t aChannel)
+{
+    ThreadError error = kThreadError_InvalidState;
+    (void)aInstance;
+
+    CORE_DECLARE_IRQ_STATE;
+    CORE_ENTER_CRITICAL();
+
+    if (sState != kStateDisabled)
+    {
+        otLogInfoPlat(sInstance, "State=kStateReceive", NULL);
+        error = kThreadError_None;
+        sState = kStateReceive;
+        setChannel(aChannel);
+        sReceiveFrame.mChannel = aChannel;
+
+        if (!sIsReceiverEnabled)
+        {
+            if (RAIL_RxStart(aChannel))
+            {
+                assert(false);
+            }
+
+            sIsReceiverEnabled = true;
+        }
+    }
+
+    CORE_EXIT_CRITICAL();
+    return error;
+}
+
+ThreadError otPlatRadioTransmit(otInstance *aInstance, RadioPacket *aPacket)
+{
+    ThreadError error = kThreadError_InvalidState;
+    RAIL_CsmaConfig_t csmaConfig = RAIL_CSMA_CONFIG_802_15_4_2003_2p4_GHz_OQPSK_CSMA;
+    RAIL_TxData_t tx_data;
+    (void)aInstance;
+
+    CORE_DECLARE_IRQ_STATE;
+    CORE_ENTER_CRITICAL();
+
+    if (sState == kStateReceive)
+    {
+        error = kThreadError_None;
+        sState = kStateTransmit;
+        sTransmitError = kThreadError_None;
+        sTransmitBusy = true;
+
+        uint8_t frame[IEEE802154_MAX_LENGTH + 1];
+        frame[0] = aPacket->mLength;
+        memcpy(frame + 1, aPacket->mPsdu, aPacket->mLength);
+
+        tx_data.dataPtr = frame;
+        tx_data.dataLength = aPacket->mLength - 1;
+        setChannel(aPacket->mChannel);
+        RAIL_RfIdleExt(RAIL_IDLE, true);
+
+        if (RAIL_TxDataLoad(&tx_data))
+        {
+            assert(false);
+        }
+
+        if (RAIL_TxStart(aPacket->mChannel, RAIL_CcaCsma, &csmaConfig))
+        {
+            assert(false);
+        }
+    }
+
+    CORE_EXIT_CRITICAL();
+    return error;
+}
+
+RadioPacket *otPlatRadioGetTransmitBuffer(otInstance *aInstance)
+{
+    (void)aInstance;
+    return &sTransmitFrame;
+}
+
+int8_t otPlatRadioGetRssi(otInstance *aInstance)
+{
+    (void)aInstance;
+    return (uint8_t)(RAIL_RxGetRSSI() / 4);
+}
+
+otRadioCaps otPlatRadioGetCaps(otInstance *aInstance)
+{
+    (void)aInstance;
+    return kRadioCapsNone;
+}
+
+bool otPlatRadioGetPromiscuous(otInstance *aInstance)
+{
+    (void)aInstance;
+    return sPromiscuous;
+}
+
+void otPlatRadioSetPromiscuous(otInstance *aInstance, bool aEnable)
+{
+    (void)aInstance;
+
+    otLogInfoPlat(sInstance, "PromiscuousMode=%d", aEnable ? 1 : 0);
```

Comment:
I don't think you need the inline if statement here, it should work automagically.